### PR TITLE
Feat/nft data input

### PR DESCRIPTION
**Pull request by haybankz**
### Description
Adds NFT file and text data input features, previews inputted data on preview screen

### CODE REVIEW

File: lib/easel_provider.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  String _fileName = "";
+  String _fileExtension = "";
+  String _fileSize = "0";
+
+  File? get file => _file;
+  String get fileName => _fileName;
+  String get fileExtension => _fileExtension;
+  String get fileSize => _fileSize;
+
+  GlobalKey<FormState> formKey = GlobalKey<FormState>();
```
Comment (kjawadDeveloper2) [inline lib/easel_provider.dart]: Move formkey from the provider. It creates issue when you do hotrestart.

File: lib/main.dart [hunk 2]
```diff
@@ -18,13 +20,18 @@ class MyApp extends StatelessWidget {
...
-        primarySwatch: Colors.blue,
+    return MultiProvider(
+      providers: [
+        ChangeNotifierProvider(create: (_) => EaselProvider()),
+      ],
+      child: MaterialApp(
+        title: 'Easel',
+        navigatorKey: navigatorKey,
+        theme: ThemeData(
+          primarySwatch: Colors.blue,
```
Comment (kjawadDeveloper2) [inline lib/main.dart]: Specify whole app theme here.

File: pubspec.yaml [hunk 3]
```diff
@@ -41,6 +41,12 @@ dependencies:
 
   flutter_svg: ^0.23.0+1
 
+  file_picker: ^4.2.0
```
Comment (kjawadDeveloper2) [inline pubspec.yaml:46]: Kindly also put details on why this dependency is used.

File: lib/screens/upload_screen.dart [hunk 4]
```diff
@@ -79,15 +90,29 @@ class _UploadWidget extends StatelessWidget {
...
+              image: MemoryImage(file!.readAsBytesSync()),
               fit: BoxFit.fill
-            )
+            ) : null
+          ),
+          child: GestureDetector(
+            onTap: ()async{
+              final result = await FileUtils().pickFile();
+              if(result != null){
+                setState(() {
```
Comment (kjawadDeveloper2) [inline lib/screens/upload_screen.dart]: No set state.

File: lib/screens/upload_screen.dart [hunk 5]
```diff
@@ -79,15 +90,29 @@ class _UploadWidget extends StatelessWidget {
...
+                  name = result.name;
+                  file = File(result.path!);
+
+                });
+              }
+
+
+            },
+              child: Image.asset("assets/icons/file.png",),
```
Comment (kjawadDeveloper2) [inline lib/screens/upload_screen.dart]: Centralize assets in one file.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]
Kindly solve these issues.

**Comment by haybankz** [comment]
@kjawadDeveloper2 issues have been solved

**Review by kjawadDeveloper2 — APPROVED** [review]
Looks good from myside.

**Review by starhelios — APPROVED** [review]

**Pull request merged**